magiclabs/magic-js
# add shouldReturnURI to configuration

### 📦 Pull Request

- When trying to implement Magic's loginWithRedirect social login, the social providers won't display redirect in an iframe
- We should return a URI optionally in this specific use case

### 🚨 Test instructions

- Create a demo app that integrates loginWithRedirect - [[link](https://magic.link/docs/api/client-side-sdks/web#loginwithredirect)](https://magic.link/docs/api/client-side-sdks/web#loginwithredirect)
    - loginWithRedirect v1
        - Ensure URI is returned on resolve
    - loginWithRedirect v2
        - Ensure URI is returned on resolve


### ⚠️ Don't forget to add a [semver](https://semver.org/) label! 
Please 🚨 **ONLY ADD ONE** 🚨 of the following labels, failing to do so may lead to adverse versioning of your changes when published:
- `minor`: New Feature

### Special Note
- intended for indiviudal use


<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @magic-ext/oauth2@9.20.0-canary.723.12288593855.0
  # or 
  yarn add @magic-ext/oauth2@9.20.0-canary.723.12288593855.0
  ```
</details>
